packages/vite/src/node/utils.ts

```diff
@@ -189,8 +189,10 @@ const knownTsOutputRE = /\.(js|mjs|cjs|jsx)$/
 export const isTsRequest = (url: string) => knownTsRE.test(cleanUrl(url))
 export const isPossibleTsOutput = (url: string) =>
   knownTsOutputRE.test(cleanUrl(url))
-export const getTsSrcPath = (filename: string) =>
-  filename.replace(/\.([cm])?(js)(x?)(\?|$)/, '.$1ts$3')
+export const getPotentialTsSrcPaths = (filename: string) => [
+  filename.replace(/\.([cm])?(js)(x?)(\?|$)/, '.$1ts$3'),
+  filename.replace(/\.([cm])?(js)(x?)(\?|$)/, '.$1tsx$3')
```

Comment:
Not sure if this is right, wouldn't it possibly give `['.mtsx', '.mtsxx']`?

---

I think adding a test case for this PR would be great too.

Reply:
Yeah actually that's a good point. The only other thing I can think of is specifically handling the `.js` case in `getTsSrcPaths`, e.g.

```js
const getTsSrcPath = (filename) => {
   const potentialSrcPaths = [filename.replace(/\.([cm])?(js)(x?)(\?|$)/, '.$1ts$3')];
   if (filename.endsWith('.js') potentialSrcPaths.push(filename.replace(/\.([cm])?(js)(x?)(\?|$)/, '.$1tsx$3'));

  return potentialSrcPaths;
}
```

Reply:
And sure, I don't mind adding a test case once the logic is good.

Reply:
Yeah, ideally when `.js` => `['.ts', '.tsx']` and `.jsx` => `['.tsx']`. The util function can be larger a bit to accommodate this, but I'd make sure it's performant too, maybe only running the regex once.